### PR TITLE
feat(BUX-207): add logger to go-paymail config

### PR DESCRIPTION
**Pull request by pawellewandowski98**
# Description

Added logger to go-paymail config

# Pull Request Checklist

- [ ] 📖 I created my PR using provided  : [CODE_STANDARDS](https://github.com/BuxOrg/bux/blob/main/.github/CODE_STANDARDS.md)
- [ ] 📖 I have read the short Code of Conduct: [CODE_OF_CONDUCT](https://github.com/BuxOrg/bux/blob/main/.github/CODE_OF_CONDUCT.md)
- [ ] 🏠 I tested my changes locally.
- [ ] ✅ I have provided tests for my changes.
- [ ] 📝 I have used conventional commits.
- [ ] 📗 I have updated any related documentation.
- [ ] 💾 PR was issued based on the Github or Jira issue.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -153,3 +153,5 @@ replace github.com/centrifugal/protocol => github.com/centrifugal/protocol v0.9.
 // Issue: go.mongodb.org/mongo-driver/x/bsonx: cannot find module providing package go.mongodb.org/mongo-driver/x/bsonx
 // Need to convert bsonx to bsoncore
 replace go.mongodb.org/mongo-driver => go.mongodb.org/mongo-driver v1.11.7
+
+//replace github.com/bitcoin-sv/go-paymail => ../../bitcoin-sv/go-paymail
```
Comment (dorzepowski) [inline go.mod:157]: Remove this

**Review by dorzepowski — CHANGES_REQUESTED** [review]

File: go.mod (hunk 1 above)
Comment (pawellewandowski98) [reply go.mod]: Done

File: client_internal.go [hunk 2]
```diff
@@ -231,6 +231,9 @@ func (c *Client) loadDefaultPaymailConfig() (err error) {
 		)
 	}
 
+	paymailLogger := c.Logger().With().Str("subservice", "go-paymail").Logger()
+	c.options.paymail.serverConfig.options = append(c.options.paymail.serverConfig.options, server.WithLogger(&paymailLogger))
```
Comment (Nazarii-4chain) [inline client_internal.go:235]: There is an error here 
`undefined: server.WithLogger`. Is it ok ?

Comment (pawellewandowski98) [reply client_internal.go:235]: I will be ok when the paymail PR will be merged, new version will come and then after update we will have this method

Comment (pawellewandowski98) [reply client_internal.go:235]: Now it works

**Review by kuba-4chain — APPROVED** [review]

**Review by dorzepowski — APPROVED** [review]

**Pull request merged**